pkg/conditions/conditions.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package conditions
+
+//
+// This class is heavily based on
+// https://github.com/knative/pkg/blob/master/apis/condition_set.go
+// but contains a more simplify approach.
+//
+// This class is intended to enable any Build CRD that might require to
+// operate and support Conditions.
+//
+// For any Build CRD that requires to operate on Conditions, they should
+// only need to implement the StatusConnect interface.
+//
+
+// StatusConnect provides access to the conditions of an
+// object that have an Status field
+type StatusConnect interface {
```

Comment:
Would `StatusConditions` be a better interface name than `StatusConnect` ? Or is a `Connect` suffix a common Go pattern ?

Reply:
not sure about the common pattern, but the `StatusConditions` is much a better name. I modified the pr.